docs/cryptodoc/src/05_09_kyber.rst

```diff
@@ -1,138 +1,299 @@
+   +-----------------------------------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------+------------------------+
+   | :ref:`Internal Keys and K-PKE <pubkey/kyber/kpke_keys>`   | :srcref:`[src/lib/pubkey/kyber/kyber_common]/kyber_keys.h`                 | Internal key class with K-PKE encryption and decryption            | 5.2, 5.3               |
+   +-----------------------------------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------+------------------------+
+   | :ref:`ML-KEM Implementation <pubkey/kyber/ml_kem_impl>`   | :srcref:`[src/lib/pubkey/kyber/ml_kem]/ml_kem_impl.h`                      | ML-KEM (internal) encapsulation and decapsulation                  | 6.2, 6.3, 7.2, 7.3     |
+   +-----------------------------------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------+------------------------+
+   | :ref:`ML-KEM <pubkey/kyber/ml_kem_api>`                   | :srcref:`[src/lib/pubkey/kyber/kyber_common]/kyber.h`                      | ML-KEM API and ML-KEM key generation                               | 7.1                    |
+   +-----------------------------------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------+------------------------+
+
+
+.. _pubkey/kyber/types:
+
+Types
+^^^^^
+
+ML-KEM employs strong types and type aliases to represent the various value
+types involved in the algorithm. This approach binds the semantic meaning of
+values to their types, resulting in a more robust interface and self-documenting
+code. Type aliases are defined for ML-KEM polynomials, polynomial vectors, and
+polynomial matrices, as well as their NTT representations. All bitstrings,
+including various hash values, random seeds, and others, are encapsulated as
+strong types. Additionally, the ML-KEM keypair and keypair seed data are
+organized within C++ structures.
+
+
+.. _pubkey/kyber/constants:
+
+Constants
+^^^^^^^^^
+
+Botan's ``KyberConstants`` class contains all parameters and constants
+outlined in Section 8 of [FIPS-203]_ (see
+:ref:`Supported ML-KEM parameter sets <pubkey_key_generation/kyber/table_params>`).
+Additionally, the class contains parameters implicitly
+derived from these constants, such as key and ciphertext sizes, along with
+various intermediate value sizes utilized within the algorithm.
+
+
+.. _pubkey/kyber/compr_helpers:
+
+Compression Helpers
+^^^^^^^^^^^^^^^^^^^
+
+Botan's helper component implements Formulas 4.7 and 4.8 from [FIPS-203]_
```

Comment:
*tries to avoid the urge to comment `Formulae` and fails even though Formulas is also correct*